src/metabase/models/permissions.clj

```diff
@@ -398,17 +406,19 @@
                              other-conditions)}]
     (when-let [revoked (t2/select-fn-set :object Permissions where)]
       (log/debug (u/format-color 'red "Revoking permissions for group %d: %s" (u/the-id group-or-id) revoked))
-      (t2/delete! Permissions where))))
+      (t2/delete! Permissions where)
+      (clear-current-user-cached-permissions!))))
 
 (defn grant-permissions!
   "Grant permissions for `group-or-id` and return the inserted permissions. Two-arity grants any arbitrary Permissions `path`.
   With > 2 args, grants the data permissions from calling [[data-perms-path]]."
   ([group-or-id path]
    (try
-     (t2/insert-returning-instances! Permissions
-                                     (map (fn [path-object]
-                                            {:group_id (u/the-id group-or-id) :object path-object})
-                                          (distinct (conj (perms.u/->v2-path path) path))))
+     (t2/insert! Permissions
+                 (map (fn [path-object]
+                        {:group_id (u/the-id group-or-id) :object path-object})
+                      (distinct (conj (perms.u/->v2-path path) path))))
```

Comment:
Not a part of this fix, but it's wasteful to use `insert-returning-instances!` when we don't actually do anything with those instances; `insert!` is much faster if we do not care about what we just inserted.